### PR TITLE
Cr 1459 rate limit eq launch

### PR DESCRIPTION
**Pull request by rpcatling**
CR-1459 describes the requirement for Rate limiting our calls to EQ launcher via the RHSvc endpoint survey-launched call.

Changes:
- added a new check method for EQ launch
- alter all IP address validation to check for IPv4 format, after discussion with @philwhiles  and @PeterBochel 
- unit tests updated to match, and some common code factored out

After discussion on the implementation, it was felt better to pass in the IP address and "load shedding modulus" and do the calculation here, so that the format validation etc can be common across all methods.

Link:
https://collaborate2.ons.gov.uk/jira/browse/CR-1459

### CODE REVIEW

File: src/main/java/uk/gov/ons/ctp/integration/ratelimiter/client/RateLimiterClient.java [hunk 1]
```diff
@@ -166,27 +172,95 @@ public void checkWebformRateLimit(Domain domain, String ipAddress)
...
     // Fail if caller doesn't meet interface requirements
     verifyArgumentSupplied("domain", domain);
-    verifyArgumentNotEmpty("ipAddress", ipAddress);
-
-    log.with("ipAddress", ipAddress).info("Check webform rate limit");
 
-    // Skip check if RHUI has not been able to get the clients IP address
-    if (ipAddress == null) {
-      log.debug("No IP Address. Not calling rate limiter");
+    if (!isValidIpAddress(ipAddress)) {
```
Comment (andrewjohnysons) [inline src/main/java/uk/gov/ons/ctp/integration/ratelimiter/client/RateLimiterClient.java:176]: loving the InetAddressValidator

File: src/test/java/uk/gov/ons/ctp/integration/ratelimiterclient/RateLimiterClientEqLaunchTest.java [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+
+@RunWith(MockitoJUnitRunner.class)
+public class RateLimiterClientEqLaunchTest extends RateLimiterClientTestBase {
+
+  @Test
+  public void shouldRejectNullDomain() {
+    CTPException exception =
+        assertThrows(
+            CTPException.class,
+            () -> {
```
Comment (andrewjohnysons) [inline src/test/java/uk/gov/ons/ctp/integration/ratelimiterclient/RateLimiterClientEqLaunchTest.java]: Don't think you need the brace - it's just an expression lamda

File: src/test/java/uk/gov/ons/ctp/integration/ratelimiterclient/RateLimiterClientEqLaunchTest.java [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+    assertTrue(exception.getMessage(), exception.getMessage().contains("'domain' cannot be null"));
+    verifyEnvoyLimiterNotCalled();
+  }
+
+  @Test
+  public void shouldRejectZeroLoadSheddingModulus() {
+    CTPException exception =
+        assertThrows(
+            CTPException.class,
+            () -> {
```
Comment (andrewjohnysons) [inline src/test/java/uk/gov/ons/ctp/integration/ratelimiterclient/RateLimiterClientEqLaunchTest.java]: And here

File: src/test/java/uk/gov/ons/ctp/integration/ratelimiterclient/RateLimiterClientFulfilmentTest.java [hunk 4]
```diff
@@ -65,25 +36,20 @@
...
-  public void setUp() {
-    simulateCircuitBreaker();
-  }
-
   @Test
   public void checkFulfilmentRateLimit_nullDomain() {
     CTPException exception =
         assertThrows(
             CTPException.class,
             () -> {
```
Comment (andrewjohnysons) [inline src/test/java/uk/gov/ons/ctp/integration/ratelimiterclient/RateLimiterClientFulfilmentTest.java]: don't need the braces - there's a few of these but it's down to style and all a bit meh

**Review by andrewjohnysons — APPROVED** [review]

File: src/test/java/uk/gov/ons/ctp/integration/ratelimiterclient/RateLimiterClientEqLaunchTest.java (hunk 2 above)
Comment (rpcatling) [reply src/test/java/uk/gov/ons/ctp/integration/ratelimiterclient/RateLimiterClientEqLaunchTest.java]: done

File: src/test/java/uk/gov/ons/ctp/integration/ratelimiterclient/RateLimiterClientEqLaunchTest.java (hunk 3 above)
Comment (rpcatling) [reply src/test/java/uk/gov/ons/ctp/integration/ratelimiterclient/RateLimiterClientEqLaunchTest.java]: done

File: src/test/java/uk/gov/ons/ctp/integration/ratelimiterclient/RateLimiterClientFulfilmentTest.java (hunk 4 above)
Comment (rpcatling) [reply src/test/java/uk/gov/ons/ctp/integration/ratelimiterclient/RateLimiterClientFulfilmentTest.java]: done

**Review by PeterBochel — APPROVED** [review]

**Pull request merged**